### PR TITLE
improve enter-git-url input

### LINKED ISSUE #811: Remove submit button in GitHub form
aint a got dang reason it needs to be there

image

### LINKED ISSUE #986: http:// URLs cause non-human-readable error
Example: https://www.drips.network/app/claim-project?projectToAdd=http%3A%2F%2Fgithub.com/github/docs

This can occur accidentally.

### PR DESCRIPTION
**Pull request by evvvritt**
closes #986
closes #811

- fixes two bugs
- removes the submit button in favor of the modal button which changes from Search to Continue if it's a valid project
- uses "ghost" Button for clear-button which has hover/focus states
- clear button appears as soon as there's text input (previously was only after submit)

Screenshot 2024-02-26 at 18 38 29
Screenshot 2024-02-26 at 18 38 53
Screenshot 2024-02-26 at 18 38 59
Screenshot 2024-02-26 at 18 39 27

### CODE REVIEW

File: src/lib/flows/claim-project-flow/steps/enter-git-url/enter-git-url.svelte [hunk 1]
```diff
@@ -208,20 +202,17 @@
...
   <svelte:fragment slot="actions">
-    <Button
-      disabled={!formValid}
-      icon={ArrowRightIcon}
-      variant="primary"
-      on:click={() => dispatch('goForward')}>Continue</Button
-    >
+    {#if formValid}
+      <Button icon={ArrowRightIcon} variant="primary" on:click={() => dispatch('goForward')}
+        >Continue</Button
```
Comment (jo-elimu) [inline src/lib/flows/claim-project-flow/steps/enter-git-url/enter-git-url.svelte:207]:
```suggestion
        >Claim</Button
```

**Review by jo-elimu — APPROVED** [review]

File: src/lib/flows/claim-project-flow/steps/enter-git-url/enter-git-url.svelte (hunk 1 above)
Comment (evvvritt) [reply src/lib/flows/claim-project-flow/steps/enter-git-url/enter-git-url.svelte:207]: Hi, thanks for your enthusiasm. If you'd like to contribute to Drips, tackling some of our ['good first issues'](https://github.com/drips-network/app/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) would be most helpful. Also, we don't merit approvals from outside contributors so please focus your energy elsewhere.

**Comment by brandonhaslegs** [comment]
I think when you enter a valid GitHub URL it should automatically "press Submit" and bring you to this step. Similar to in the Drip List creator, when you enter a GitHub URL it automatically adds it to the list.

image

**Comment by evvvritt** [comment]
Sure, it now submits the input on paste if it's valid (doing it as you type doesn't work / fires immediately)

**Comment by evvvritt** [comment]
can I get a new review @brandonhaslegs / @efstajas / @jtourkos :}

**Comment by evvvritt** [comment]
What should the design of the focus state be then? It's gonna get focused and before using our ghost-button there was no visual indicator

**Comment by brandonhaslegs** [comment]
We have a X button in the global search box that doesn't have this focus state. I'd hope it can just work like that.

image

**Comment by evvvritt** [comment]
Ok, doesn't focus now like the global search input.

**Comment by efstajas** [comment]
love the auto-submit on paste, but it's possible now to edit the text field after searching & finding a project. Look at this state:

Screenshot 2024-03-06 at 10 03 09

I entered a URL to repo #1, then clicked "Search", then replaced the URL with another repo URL, and now it's not clear how to "Search" for the new repo URL. I might even expect that when I click "Continue" that it selects the newly-entered URL, which it doesn't. Especially because the "Continue" button replaces the "Search" button.

IMO it should make the text field non-editable when a repo is selected, and display a "Clear" button next to "Continue".

**Comment by brandonhaslegs** [comment]
Is it possible we can only support pasting a URL in. I don't see anyone typing in a URL here. So we disable typing in the field and only allow pasting and clearing the field?

**Comment by evvvritt** [comment]
Idk, I think it's fine? Most people will paste but if decide to type why block them?

I reverted to your scheme @efstajas (disabled on valid submit).

**Review by efstajas — APPROVED** [review]
✨

**Pull request merged**